feat: add autoexposure and whitebalance configuration

higor-melo opened the pull request:
# What is this PR for
<!-- A clear description of what this PR do and the changes made.
If you want you can also add the shortcut link here -->
- [x] [Shortcut](https://app.shortcut.com/tidewise/story/41013/add-missing-control-settings-to-the-lucid-component)

The lucid cameras were under performing in dark scenarios, these configurations enable a wider number of configuration parameters for both the auto-exposure and the white balance.
As for the white balance, the lucid cameras have a NIR filter and despite it's upsides we have a major downside regarding night time operations, it makes the image darker. With the removal of the filter the colors were too red, hence the need to re-configure the white balance.
Finally, i've also discovered a bug involving target brightness which is better explained on the commit that fixes it.

# How I did it
<!-- Explain a little bit of your implementation -->
Added new configuration options to the driver by altering values on the nodes.

# Results, How I tested
<!-- Explain how you tested you PR, if there is a visual output,
     a GIF or image is welcome -->
Empyrical testing with the camera on the office.
Test using the online_test.rb

# Checklist
- [x] Assign yourself  in the PR
- [x] Write unit tests (when relevant)

higor-melo committed:
feat: add autoExposure parameters

Added feedback from meanExposure time
Added autoExposureDamping control
Added autoExposureAlgorithm control

higor-melo committed:
feat: add balanceConfiguration task

Right now task is responsible for configuring
the WhiteBalanceEnable, WhiteBalanceAuto
and WhiteBalanceAnchorSelector.
Normally those would be gathered inside AnalogConfig
but it would make the method too large.

higor-melo committed:
feat: add balanceRatioSelector enum and config

higor-melo committed:
feat: add balanceRatio variable

higor-melo committed:
feat: add balanceRatioSelector configuration and fix average_exposure

higor-melo committed:
feat: add balance_ratio control

Fixed Typos, moved auto mode configuration out of if statement
since both white_balance_enable and white_auto_mode must
be off to allow writing of balance_ratio values

higor-melo committed:
fix: exposureAutoDamping is now a floatPtr

higor-melo committed:
feat: add check for current_algorithm value

Also fixed name from exposure_auto_algorithm vector

higor-melo committed:
fix: added condition to exposure lower and upper limit

When ShortExposureEnable is TRUE, lower and upper limits
turn into READ-ONLY

higor-melo committed:
chore: add commentary explaining the short_exposure_state config

Commentary and Linting

higor-melo committed:
fix: change BalanceWhiteAutoAnchorSelector to Default

higor-melo committed:
fix: configuration of target brightness

While testing the new features implemented
i noticed that target brightness cannot be
set when exposure auto mode is set to
EXPOSURE_AUTO_ONCE

higor-melo committed:
chore: altered tests and added commentary

This tests must be run in a different fashion
since they're live. Created a little tutorial
inside CMakeLists.txt

higor-melo committed:
feat: add current balance ratio feedback

higor-melo committed:
fix: sets multiple balance ratios for RGB

Added new variables, removed unused variables
Fixed if check

doudou commented on camera_lucidTypes.hpp:
```diff
@@ -209,8 +261,18 @@ namespace camera_lucid {
+        ExposureAutoAlgorithm exposure_auto_algorithm = EXPOSURE_AUTO_ALGORITHM_MEAN;
+        /** Sets the automatic exposure damping represented as %.
+         * Maximum, Minimum: 99.6094, 0.390625 */
+        double exposure_auto_damping = 89.8438;
```
I think that it's probably camera-dependent. I'd prefer that the default is `base::unknown<double>()` (i.e. NaN) and that you do not set the value if it is nan (plus update comment)

doudou commented on tasks/Task.cpp:
```diff
@@ -1022,6 +1169,10 @@ void Task::collectInfo()
                 Arena::GetNodeValue<double>(m_device->GetNodeMap(), "DeviceTemperature"));
+            info_message.average_exposure =
+                Arena::GetNodeValue<int64_t>(m_device->GetNodeMap(), "CalculatedMean");
+            info_message.current_balance_ratio =
+                Arena::GetNodeValue<double>(m_device->GetNodeMap(), "BalanceRatio");
```
Shouldn't be reading the three channels here ?

higor-melo committed:
fix: add exposure_auto_damping value check

Now it doesn't set exposure_auto_damping if it
isn't set on the .yml
Fixed min_exposure_time conversion
Removed current_balance_ratio

higor-melo commented:
It's ready to merge @doudou , just waiting your approval.
